Assertion error when deleting last emitter in ParticleFX

**Describe the bug (REQUIRED)**
Assertion bug when trying to delete the **last** emitter
https://drive.google.com/file/d/1MGFOZtNWAVQ5-xQSvztygq87IrkX_hGr/view?usp=drive_link
screenshot

**To Reproduce (REQUIRED)**
Steps to reproduce the behavior:
1. Create new project
2. Create a particlefx
3. Delete the emitters
It caused the bugs
For more info: https://drive.google.com/file/d/1NYm2FRxJXr_a4ib3W3oJxZKUShkorqri/view?usp=drive_link
video

**Expected behavior (REQUIRED)**
It should just say Not Allowed or just let it delete

**Defold version (REQUIRED):**
  - Version defold 1.12.4.402218d-1

**Platforms (REQUIRED):**
 - Platforms: Local build
 - OS: ArchLinux

**Minimal repro case project (OPTIONAL):**
[Please attach a minimal project where the bug is reproduced. This will greatly help the person trying to investigate and fix the bug.](https://drive.google.com/drive/folders/1Cu9SYZQHrT_q93rcfN6Xor1FRlDmO0qU?usp=drive_link)

**Workaround (OPTIONAL):**
Just don't delete it 

**Screenshots (OPTIONAL):**
https://drive.google.com/file/d/1MGFOZtNWAVQ5-xQSvztygq87IrkX_hGr/view?usp=drive_link

```
java.lang.AssertionError: Assert failed: (vector? items)
	at editor.localization$annotate_as_sorted.invokeStatic(localization.clj:647)
	at editor.localization$annotate_as_sorted.invoke(localization.clj:647)
	at editor.particlefx$fn__121643.invokeStatic(particlefx.clj:1164)
	at editor.particlefx$fn__121643.invoke(particlefx.clj:1153)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:33)
	at clojure.lang.RestFn.invoke(RestFn.java:411)
	at clojure.lang.Var.invoke(Var.java:386)
	at editor.particlefx$ParticleFXNode$behavior$node_outline$fn__121768.invoke(particlefx.clj:1114)
	at internal.node$trace_expr.invokeStatic(node.clj:46)
	at internal.node$trace_expr.invoke(node.clj:35)
	at editor.particlefx$ParticleFXNode$behavior$node_outline.invokeStatic(particlefx.clj:1114)
	at editor.particlefx$ParticleFXNode$behavior$node_outline.invoke(particlefx.clj:1114)
	at clojure.lang.Var.invoke(Var.java:395)
	at internal.node.NodeImpl.produce_value(node.clj:338)
	at internal.node$pull_first_input_value.invokeStatic(node.clj:1229)
	at internal.node$pull_first_input_value.invoke(node.clj:1224)
	at editor.app_view$AppView$behavior$active_outline$fn__104449.invoke(app_view.clj:350)
	at internal.node$trace_expr.invokeStatic(node.clj:46)
	at internal.node$trace_expr.invoke(node.clj:35)
	at editor.app_view$AppView$behavior$active_outline.invokeStatic(app_view.clj:350)
	at editor.app_view$AppView$behavior$active_outline.invoke(app_view.clj:350)
	at clojure.lang.Var.invoke(Var.java:395)
	at internal.node.NodeImpl.produce_value(node.clj:338)
	at internal.node$pull_first_input_value.invokeStatic(node.clj:1229)
	at internal.node$pull_first_input_value.invoke(node.clj:1224)
	at editor.outline_view$OutlineView$behavior$pane_desc$fn__110993.invoke(outline_view.clj:289)
	at internal.node$trace_expr.invokeStatic(node.clj:46)
	at internal.node$trace_expr.invoke(node.clj:35)
	at editor.outline_view$OutlineView$behavior$pane_desc.invokeStatic(outline_view.clj:289)
	at editor.outline_view$OutlineView$behavior$pane_desc.invoke(outline_view.clj:289)
	at clojure.lang.Var.invoke(Var.java:395)
	at internal.node.NodeImpl.produce_value(node.clj:338)
	at internal.node$pull_first_input_value.invokeStatic(node.clj:1229)
	at internal.node$pull_first_input_value.invoke(node.clj:1224)
	at editor.app_view$AppView$behavior$outline_active$fn__104414.invoke(app_view.clj:350)
	at internal.node$trace_expr.invokeStatic(node.clj:46)
	at internal.node$trace_expr.invoke(node.clj:35)
	at editor.app_view$AppView$behavior$outline_active.invokeStatic(app_view.clj:350)
	at editor.app_view$AppView$behavior$outline_active.invoke(app_view.clj:350)
	at clojure.lang.Var.invoke(Var.java:395)
	at internal.node.NodeImpl.produce_value(node.clj:338)
	at editor.app_view$AppView$behavior$right_split_desc$fn__104370.invoke(app_view.clj:350)
	at internal.node$trace_expr.invokeStatic(node.clj:46)
	at internal.node$trace_expr.invoke(node.clj:35)
	at editor.app_view$AppView$behavior$right_split_desc.invokeStatic(app_view.clj:350)
	at editor.app_view$AppView$behavior$right_split_desc.invoke(app_view.clj:350)
	at clojure.lang.Var.invoke(Var.java:395)
	at internal.node.NodeImpl.produce_value(node.clj:338)
	at internal.node$node_value.invokeStatic(node.clj:479)
	at internal.node$node_value.invoke(node.clj:470)
	at dynamo.graph$node_value.invokeStatic(graph.clj:1259)
	at dynamo.graph$node_value.invoke(graph.clj:1234)
	at editor.app_view$refresh_right_split_BANG_.invokeStatic(app_view.clj:2191)
	at editor.app_view$refresh_right_split_BANG_.invoke(app_view.clj:2189)
	at editor.app_view$make_app_view$fn__106703.invoke(app_view.clj:2228)
	at editor.ui$__GT_timer$fn__33435$fn__33436.invoke(ui.clj:2378)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1990)
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1990)
	at clojure.lang.RestFn.invoke(RestFn.java:428)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:135)
	at clojure.core$apply.invokeStatic(core.clj:671)
	at clojure.core$bound_fn_STAR_$fn__5837.doInvoke(core.clj:2020)
	at clojure.lang.RestFn.invoke(RestFn.java:400)
	at clojure.lang.AFn.run(AFn.java:22)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$0(PlatformImpl.java:424)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$0(GtkApplication.java:240)
	at java.base/java.lang.Thread.run(Thread.java:1474)

```

--- a/editor/src/clj/editor/particlefx.clj
+++ b/editor/src/clj/editor/particlefx.clj
@@ -1155,8 +1155,8 @@
             ;; should never change
             (let [basis (:basis _evaluation-context)
                   outlines (group-by #(g/node-instance? basis ModifierNode (:node-id %)) child-outlines)
-                  mod-outlines (get outlines true)
-                  emitter-outlines (get outlines false)]
+                  mod-outlines (get outlines true [])
+                  emitter-outlines (get outlines false [])]
               {:node-id _node-id
                :node-outline-key "ParticleFX"
                :label (localization/message "outline.particlefx")

--- a/editor/test/integration/particlefx_test.clj
+++ b/editor/test/integration/particlefx_test.clj
@@ -97,6 +97,14 @@
         (test-util/with-prop [emitter :animation v]
           (is (g/error? (test-util/prop-error emitter :animation))))))))
 
+(deftest delete-last-emitter
+  (test-util/with-loaded-project
+    (let [node-id (test-util/resource-node project "/particlefx/default.particlefx")
+          emitter (:node-id (test-util/outline node-id [0]))]
+      (g/delete-node! emitter)
+      (let [outline (g/node-value node-id :node-outline)]
+        (is (= [] (:children outline)))))))
+
 (deftest particle-scene
   (test-util/with-loaded-project
     (let [node-id (project/get-resource-node project "/particlefx/default.particlefx")

